src/meta/src/hummock/manager/mod.rs

```diff
@@ -201,6 +203,42 @@ where
         Ok(instance)
     }
 
+    pub async fn start_compaction_heartbeat(
+        hummock_manager: Arc<Self>,
+    ) -> (JoinHandle<()>, Sender<()>) {
+        let (shutdown_tx, mut shutdown_rx) = tokio::sync::oneshot::channel();
+        let compactor_manager = hummock_manager.compactor_manager.clone();
+        let join_handle = tokio::spawn(async move {
+            let mut min_interval = tokio::time::interval(std::time::Duration::from_millis(1000));
+            loop {
+                tokio::select! {
+                    // Wait for interval
+                    _ = min_interval.tick() => {
+                    },
+                    // Shutdown
+                    _ = &mut shutdown_rx => {
+                        tracing::info!("Compaction heartbeat checker is stopped");
+                        return;
+                    }
+                }
+                for (context_id, mut task) in compactor_manager.get_expired_tasks() {
+                    if let Some(compactor) = compactor_manager.get_compactor(context_id) {
+                        let _ = compactor.cancel_task(task.task_id).await;
+                    }
+
+                    // Change task status to failed
+                    task.task_status = false;
+
+                    if let Err(e) = hummock_manager.report_compact_task(context_id, &task).await {
```

Comment:
Should use cancel_compact_task instead of report_compact_task.

report_compact_task(context_id, &task) will fail if context_id is not a valid member of the cluster.